drivers/scheduler/rke/rke.go

```diff
@@ -1,28 +1,270 @@
+func (r *Rancher) Init(scheduleOpts scheduler.InitOptions) error {
+	var err error
+	err = r.K8s.Init(scheduleOpts)
+	RancherClusterParametersValue, err = r.GetRancherClusterParametersValue()
+	if err != nil {
+		return err
+	}
+	rancherClientOpts := rancherClientBase.ClientOpts{
+		URL:       RancherClusterParametersValue.Endpoint,
+		TokenKey:  RancherClusterParametersValue.Token,
+		AccessKey: RancherClusterParametersValue.AccessKey,
+		SecretKey: RancherClusterParametersValue.SecretKey,
+		Insecure:  true,
+	}
+	r.client, err = rancherClient.NewClient(&rancherClientOpts)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetRancherClusterParametersValue returns the rancher token, endpoint, secret key, access key
+func (r *Rancher) GetRancherClusterParametersValue() (*RancherClusterParameters, error) {
+	var rkeData map[string]interface{}
+	var rkeParameters RancherClusterParameters
+	// TODO Rancher URL for cloud cluster will not be fetched from master node IP
+	masterNodeName := node.GetMasterNodes()[0].Name
+	endpoint := "https://" + masterNodeName + "/v3"
+	rancherURL := "https://" + masterNodeName + "/v3-public/localProviders/local?action=login"
+	//TODO: Get the values from config map after https://github.com/portworx/torpedo/pull/1517 is merged
+	username := "admin"
+	password := "1JVWD8juKPB8fVgK"
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+	body := strings.NewReader(fmt.Sprintf(`{"username":"%s","password":"%s"}`, username, password))
+	req, err := http.NewRequest("POST", rancherURL, body)
+	if err != nil {
+		log.Errorf("Failed to create rancher POST request:", err)
+		return &rkeParameters, err
+	}
+	req.Header.Add("Content-Type", "application/json")
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Infof("Failed to send request:", err)
+		return &rkeParameters, err
+	}
+	defer resp.Body.Close()
+	respBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Infof("Failed to read response:", err)
+		return &rkeParameters, err
+	}
+	response := string(respBody)
+	err = json.Unmarshal([]byte(response), &rkeData)
+	if err != nil {
+		log.Infof("Error:", err)
+		return &rkeParameters, err
+	}
+	rkeParameters.Endpoint = endpoint
+	rkeToken := rkeData["token"].(string)
+	if len(strings.Split(rkeToken, ":")) != 2 {
+		return &rkeParameters, fmt.Errorf("invalid RKE token value")
+	}
+	rkeParameters.Token = rkeToken
+	rkeParameters.AccessKey = strings.Split(rkeToken, ":")[0]
+	rkeParameters.SecretKey = strings.Split(rkeToken, ":")[1]
+	return &rkeParameters, err
+}
+
+// GetActiveRancherClusterID returns the ID of active rancher cluster
+func (r *Rancher) GetActiveRancherClusterID(clusterName string) (string, error) {
+	var clusterId string
+	clusterCollection, err := r.client.Cluster.List(nil)
+	if err != nil {
+		return "", err
+	}
+	for _, cluster := range clusterCollection.Data {
+		if cluster.Name == clusterName {
```

Comment:
How about returning an error in case clusterName is not found?

Reply:
done